test/ReaderTest.cpp

```diff
@@ -357,26 +359,49 @@ BOOST_AUTO_TEST_CASE(TestGetSubBlockNames)
         m_test_indexed_block
     */
 
-   std::vector<std::pair<std::string, unsigned int>> expected_subblocks = {
-      {"m_test_block", 0},
-      {"m_nested_block", 1},
-      {"m_test_nested_indexed_block", 2},
-      {"m_test_block", 1},
-      {"m_test_repeated_block", 1},
-      {"m_test_indexed_block", 1},
-      {schrodinger::mae::ATOM_BLOCK, 0},
-      {schrodinger::mae::BOND_BLOCK, 0},
-   };
-   std::vector<std::pair<std::string, unsigned int>> actual_subblocks;
-   write_block_names(*b, 0, actual_subblocks);
-
-   BOOST_REQUIRE(actual_subblocks.size() == expected_subblocks.size());
-   for (unsigned int i = 0; i < actual_subblocks.size(); ++i) {
-      auto actual = actual_subblocks[i];
-      auto expected = expected_subblocks[i];
-      BOOST_CHECK_EQUAL(actual.first, expected.first);
-      BOOST_CHECK_EQUAL(actual.second, expected.second);
-   }
+    std::vector<std::pair<std::string, unsigned int>> expected_subblocks = {
+        {"m_test_block", 0},
+        {"m_nested_block", 1},
+        {"m_test_nested_indexed_block", 2},
+        {"m_test_block", 1},
+        {"m_test_repeated_block", 1},
+        {"m_test_indexed_block", 1},
+        {schrodinger::mae::ATOM_BLOCK, 0},
+        {schrodinger::mae::BOND_BLOCK, 0},
+    };
+    std::vector<std::pair<std::string, unsigned int>> actual_subblocks;
+    write_block_names(*b, 0, actual_subblocks);
+
+    BOOST_REQUIRE(actual_subblocks.size() == expected_subblocks.size());
+    for (unsigned int i = 0; i < actual_subblocks.size(); ++i) {
+        auto actual = actual_subblocks[i];
+        auto expected = expected_subblocks[i];
+        BOOST_CHECK_EQUAL(actual.first, expected.first);
+        BOOST_CHECK_EQUAL(actual.second, expected.second);
+    }
 }
 
+BOOST_AUTO_TEST_CASE(TestParsingPropertyNameWithColon)
+{
+    auto ss = std::make_shared<std::stringstream>();
+    *ss << "\n"
```

Comment:
This would probably be more readable as either a multi-line string literal or as multiple consecutive string literals (which are automatically concatenated by the compiler)

    ss << "\n"
          "\n"
          "f_m_ct {\n"

or

    ss << R"(

f_m_ct {

s_m_prop:name::with:::many::::colons
:::
1.1.0
}
");

Reply:
good point